xpdacq/utils.py

```diff
@@ -92,39 +93,79 @@ def export_userScriptEtc():
-    moved_f_list = []
+    # unpack every archived file in Import/
     for f in f_list:
-        full_path = os.path.join(src_dir, f)
-        (root, ext) = os.path.splitext(f)
-        if ext == '.yml':
-            shutil.copy(full_path, dst_dir)
-            moved_f_list.append(f)
-            # FIXME - do we want user confirmation?
-            os.remove(full_path)
-        else:
-            try:
-                shutil.unpack_archive(full_path, dst_dir)
-                moved_f_list.append(f)
-                # FIXME - do we want user confirmation?
-                os.remove(full_path)
-            except ReadError:
-                print('Unrecongnized file type {} is found inside {}'.format(f, src_dir))
+        try:
+            # shutil should handle all compressed cases
+            tar_full_path = os.path.join(src_dir, f)
+            shutil.unpack_archive(tar_full_path, src_dir)
+        except ReadError:
+            pass
+    f_list = os.listdir(src_dir) # new f_list, after unpack
+    moved_list = []
+    failure_list = []
+    for f_name in f_list:
+        if os.path.isfile(os.path.join(src_dir, f_name)):
+            src_full_path = os.path.join(src_dir, f_name)
+            (root, ext) = os.path.splitext(f_name)
+            if ext == '.yml':
+                dst_dir = glbl.yaml_dir
+                yml_dst_name = _copy_and_delete(f_name, src_full_path, dst_dir)
+                moved_list.append(yml_dst_name)
+            elif ext == '.py':
+                dst_dir = glbl.usrScript_dir
+                py_dst_name = _copy_and_delete(f_name, src_full_path, dst_dir) 
+                moved_list.append(py_dst_name)
+            elif ext == '.npy':
+                dst_dir = glbl.config_base
+                npy_dst_name = _copy_and_delete(f_name, src_full_path, dst_dir) 
+                moved_list.append(npy_dst_name)
+            elif ext in ('.tar', '.zip', '.gztar'):
                 pass
-    return moved_f_list
+            else:
+                print('{} is not a supported format'.format(f_name))
+                failure_list.append(f_name)
+                pass
+        else:
+            # don't expect user to have directory
+            print('Expect a file but get a directory {}. Did you properly archive it?'.format(f_name))
+            failure_list.append(f_name)
+            pass
+    if failure_list:
+        print('Finished importing. Failed to move {} but they will leave in Import/'.format(failure_list))
+    return moved_list
+
+def _copy_and_delete(f_name, src_full_path, dst_dir):
+    shutil.copy(src_full_path, dst_dir)
+    dst_name = os.path.join(dst_dir, f_name) 
+    if os.path.isfile(dst_name):
+        print('{} has been successfully moved to {}'.format(f_name, dst_dir))
+        os.remove(src_full_path)
+        return dst_name
+    else:
+        print('We have problem moving {} it will still leave at xpdUser/Import/'.format(f_name))
```

Comment:
We had a problem moving {}.  Most likely it is not a supported file type (e.g., .yml, .py, .npy, .tar, .gz).  It will not be available for use in xpdAcq, but it will be left in the xpdUser/Import/ directory'